web/modules/custom/joinup_federation/src/Plugin/pipeline/Step/RemoveUnsupportedData.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\joinup_federation\Plugin\pipeline\Step;
+
+use Drupal\joinup_federation\JoinupFederationStepPluginBase;
+use Drupal\rdf_entity\Entity\Query\Sparql\SparqlArg;
+use Drupal\rdf_entity\Entity\RdfEntityMapping;
+use Drupal\rdf_entity\RdfEntityGraphStoreTrait;
+
+/**
+ * Defines a process step that removes the triples not supported by Joinup.
+ *
+ * Scan the imported triples (which are now in the sink graph) and filter out
+ * all that are not Joinup entities, as collections, solutions, releases,
+ * distributions, licences, owners or contact information.
+ *
+ * @PipelineStep(
+ *  id = "remove_unsupported_data",
+ *  label = @Translation("Remove data not supported by Joinup"),
+ * )
+ */
+class RemoveUnsupportedData extends JoinupFederationStepPluginBase {
+
+  use RdfEntityGraphStoreTrait;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function execute(array &$data) {
+    $graph_uri = $this->getGraphUri('sink');
+
+    $rdf_entity_bundle_uris = [];
+    /** @var \Drupal\rdf_entity\RdfEntityMappingInterface $mapping */
+    foreach (RdfEntityMapping::loadMultiple() as $mapping) {
+      // Only add rdf:type URI for RDF entities.
+      if ($mapping->getTargetEntityTypeId() === 'rdf_entity') {
+        $rdf_entity_bundle_uris[] = $mapping->getRdfType();
+      }
+    }
+    $rdf_entity_bundle_uris = SparqlArg::serializeUris($rdf_entity_bundle_uris);
+
+    $query = <<<Ouery
+DELETE FROM <$graph_uri> {
+  ?subject ?predicate ?object
+}
+WHERE {
+  ?subject ?predicate ?object
```

Comment:
Would this produce the same result?
DELETE FROM <$graph_uri> {
  ?subject ?predicate ?object
}
WHERE {
  ?subject ?predicate ?object .
  ?subject a ?entity_type .
  FILTER ( ?entity_type NOT IN ( $rdf_entity_bundle_uris ) ) .
}